7.js

```diff
@@ -0,0 +1,61 @@
+const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+'Jul', 'Aug', 'Sep', 'Oct', 'Nov',
+'Dec']
+
+function formatDate (date) {
+  if (date.includes('-')) {
+    return date
+  } else if (date.includes('/')) {
+    const dateArray = date.split('/')
+    return `${fullYear(dateArray[2])}-${dateArray[0]}-${dateArray[1]}`
+  } else if (date.includes('#')) {
+    const dateArray = date.split('#')
+    return `${fullYear(dateArray[1])}-${dateArray[0]}-${dateArray[2]}`
+  } else if (date.includes('*')) {
+    const dateArray = date.split('*')
+    return `${dateArray[2]}-${dateArray[1]}-${dateArray[0]}`
+  } else {
+    months.forEach(month => {
+      if (!date.includes(month)) {
+        return 'Date format not recognized'
+      }
+    })
+
+    const dateArray = date.split(' ')
+    dateArray[1] = dateArray[1].slice(0, 2)
+
+    if (dateArray[2].length === 2) {
+      dateArray[2] = fullYear(dateArray[2])
+    }
+
+    dateArray[0] = monthToNumber(dateArray[0])
+
+    return `${dateArray[2]}-${dateArray[0]}-${dateArray[1]}`
+  }
+}
+
+const fullYear = year => year < 50 ? `20${year}` : `19${year}`
+
+const monthToNumber = month => months.indexOf(month) < 9
```

Comment:
This formatting is a little confusing.  I recommend avoiding the use of a ternary operator to promote readability and clarity in the code:
```js
const monthNumber = months.indexOf( month ) + 1

if( months.indexOf( month ) < OCTOBER ) {
  return `0${monthNumber}`
} else {
  return monthNumber
}
```
Note that I also removed the "magic number" 9, and replaced it with a constant that provides a little more clarity in the code.